File: app/src/main/java/com/example/notes/data/locale/dao/NoteDAO.java
//package com.example.notes.data.locale.dao;
//
//import androidx.lifecycle.LiveData;
//import androidx.room.Dao;
//import androidx.room.Delete;
//import androidx.room.Insert;
//import androidx.room.OnConflictStrategy;
//import androidx.room.Query;
//import androidx.room.Update;
//
//import com.example.notes.data.locale.entity.Note;
//
//import java.util.List;
//
//@Dao
//public interface NoteDAO {
//    @Query("SELECT * FROM NoteEntityKotlin ORDER BY data")
//    LiveData<List<Note>> listOfAllNotesLiveData();
//
//    @Query("SELECT * FROM NoteEntityKotlin ORDER BY data")
//    List<Note> listOfAllNoteWithoutLiveData();
//
//    @Query("SELECT * FROM NoteEntityKotlin WHERE nameGroup == :nameGroup ORDER BY data")
//    List<Note> listOfAllNoteByNameGroup(String nameGroup);
//
//    @Query("SELECT DISTINCT nameGroup FROM NoteEntityKotlin WHERE nameGroup !=''  ORDER BY data")
//    LiveData<List<String>> listOfAllGroupsName();
//
//    @Query("SELECT * FROM NoteEntityKotlin WHERE id = :id")
//    Note noteById(int id);
//
//    @Insert
//    void insertNote(Note noteEntityKotlinEntry);
//
//    @Update
//    void updateNote(Note noteEntityKotlinEntry);
//
//    @Update(onConflict = OnConflictStrategy.REPLACE)
//    void updateNoteList(List<Note> noteEntityKotlinEntry);
//
//    @Delete
//    void deleteNote(Note noteEntityKotlinEntry);
//
//    @Delete
//    void deleteNotes(List<Note> noteEntityKotlinEntry);
//
//}
//
//

File: data/src/main/java/com/example/data/local/dao/NoteDao.kt
package com.example.data.local.dao

import androidx.room.*
import com.example.domainn.entity.NoteEntity
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Single
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.StateFlow

@Dao
interface NoteDao {

    @Query("SELECT * FROM notes_table ORDER BY sortDate DESC")
    fun getAllNotes(): Single<List<NoteEntity>>

    @Query("SELECT * FROM notes_table WHERE nameGroup == :nameGroup ORDER BY createDate DESC")
    fun getAllNotesByGroup(nameGroup: String): Single<List<NoteEntity>>

    @Query("SELECT DISTINCT nameGroup FROM notes_table WHERE nameGroup !='' ORDER BY createDate")
    fun getAllNotesWithNameGroup(): Single<List<String>>

    @Query("SELECT * FROM notes_table WHERE id == :id")
    fun getNoteById(id: Int): Single<NoteEntity>

    @Query("SELECT * FROM notes_table WHERE id == :id")
    fun getNoteByIdFlow(id: Int): Flow<NoteEntity>

    @Insert
    fun addNote(noteEntity: NoteEntity): Completable

    @Update
    fun updateNote(noteEntity: NoteEntity): Completable

    @Query("DELETE FROM notes_table WHERE id = :id")
    fun deleteNote(id: Int): Completable

    @Delete
    fun deleteNotes(notes: List<NoteEntity>): Completable
}
